cluster-api/src/main/java/com/sequenceiq/cloudbreak/cluster/api/ClusterSetupService.java

```diff
@@ -56,7 +56,7 @@ void configureManagementServices(TemplatePreparationObject templatePreparationOb
 
     ExtendedPollingResult waitForHosts(Set<InstanceMetadataView> hostsInCluster) throws ClusterClientInitException;
 
-    void waitForHostsHealthy(Set<InstanceMetadataView> hostsInCluster) throws ClusterClientInitException;
+    List<InstanceMetadataView> waitForHostsHealthy(Set<InstanceMetadataView> hostsInCluster) throws ClusterClientInitException;
```

Comment:
Have this return ExtendedPollingResult to be consistent with other APIs ?, and the invoker can decide how best to handle the failure.